presto-main/src/main/java/io/prestosql/memory/QueryContext.java

```diff
@@ -70,6 +70,10 @@
     private final SpillSpaceTracker spillSpaceTracker;
     private final Map<TaskId, TaskContext> taskContexts = new ConcurrentHashMap<>();
 
+    @GuardedBy("this")
+    private boolean isResourceOverCommit;
+    private boolean memoryLimitsInitialized;
```

Comment:
rename to `memoryLimitInitialized`

Reply:
There are multiple limits involved (`maxUserMemory` / `maxTotalMemory`).